src/storage/test/QueryBoundBenchmark.cpp

```diff
@@ -0,0 +1,186 @@
+                }
+            }
+            // Generate 5 in-edges for each edgeType, the edgeType is negative
+            for (auto srcId = 20001; srcId <= 20005; srcId++) {
+                VLOG(3) << "Write part " << partId << ", vertex " << vertexId << ", src " << srcId;
+                for (auto version = 0; version < 3; version++) {
+                    auto key = NebulaKeyUtils::edgeKey(partId, vertexId, -101,
+                                                 srcId - 20001, srcId,
+                                                 std::numeric_limits<int>::max() - version);
+                    data.emplace_back(std::move(key), "");
+                }
+            }
+        }
+        kv->asyncMultiPut(
+            0, partId, std::move(data),
+            [&](kvstore::ResultCode code) {
+                CHECK_EQ(code, kvstore::ResultCode::SUCCEEDED);
+            });
+    }
+}
+
+void setUp(const char* path) {
+    gKV = TestUtils::initKV(path);
+    schema.reset(new storage::AdHocSchemaManager());
+    schema->addEdgeSchema(
+        0 /*space id*/, 101 /*edge type*/, TestUtils::genEdgeSchemaProvider(10, 10));
+    for (auto tagId = 3001; tagId < 3010; tagId++) {
+        schema->addTagSchema(
+            0 /*space id*/, tagId, TestUtils::genTagSchemaProvider(tagId, 3, 3));
+    }
+    mockData(gKV.get());
+}
+
+cpp2::GetNeighborsRequest buildRequest(bool outBound = true) {
+    cpp2::GetNeighborsRequest req;
+    req.set_space_id(0);
+    decltype(req.parts) tmpIds;
+    for (auto partId = 0; partId < FLAGS_req_parts; partId++) {
+        for (auto vertexId = 1; vertexId <= FLAGS_vrpp; vertexId++) {
+            tmpIds[partId].push_back(vertexId);
+        }
+    }
+    req.set_parts(std::move(tmpIds));
+    req.set_edge_type(outBound ? 101 : -101);
+    // Return tag props col_0, col_2, col_4
+    decltype(req.return_columns) tmpColumns;
+    for (int i = 0; i < 3; i++) {
+        tmpColumns.emplace_back(
+            TestUtils::propDef(cpp2::PropOwner::SOURCE,
+                               folly::stringPrintf("tag_%d_col_%d", 3001 + i*2, i*2),
+                               3001 + i*2));
+    }
+    tmpColumns.emplace_back(TestUtils::propDef(cpp2::PropOwner::EDGE,
+                                               folly::stringPrintf("_dst")));
+    tmpColumns.emplace_back(TestUtils::propDef(cpp2::PropOwner::EDGE,
+                                               folly::stringPrintf("_rank")));
+    // Return edge props col_0, col_2, col_4 ... col_18
+    for (int i = 0; i < 10; i++) {
+        tmpColumns.emplace_back(
+            TestUtils::propDef(cpp2::PropOwner::EDGE,
+                               folly::stringPrintf("col_%d", i*2)));
+    }
+    req.set_return_columns(std::move(tmpColumns));
+    return req;
+}
+
+}  // namespace storage
+}  // namespace nebula
+
+void run(int32_t iters, int32_t handlerNum) {
+    FLAGS_max_handlers_per_req = handlerNum;
+    auto req = nebula::storage::buildRequest();
```

Comment:
buildRequest seems time consuming. separate to timing.

Reply:
Good point.